specs/error_state/error_write_protection.md

```diff
@@ -0,0 +1,25 @@
+# ErrorWriteProtection state
+
+## Procedure
+there are some op codes which modify state. there are `[SSTORE, CREATE, CREATE2, 
+CALL, SELFDESTRUCT, LOG0, LOG1, LOG2, LOG3, LOG4]`
+when execution call context is ready only (static call), these op codes running will encounter
+write protection error.
+### EVM behavior
+in above op codes which modify state, `CALL` is somewhat special. for non call op codes(SSTORE, CREATE, etc)
+first check if running in ready only call context, if yes, throw write protection error.
+for `CALL` op code, it will also check `value` is not zero. only both non zero value & ready only call context, then
+throw write protection error.
+
+### Constraints
+1. constrain this error happens in one of op codes `[SSTORE, CREATE, CREATE2, 
+CALL, SELFDESTRUCT, LOG0, LOG1, LOG2, LOG3, LOG4]`
+2. current call context must be readonly & internal call(since requires at least one `staticcall` earlier).
```

Comment:
i am not sure whether "& internal call" is needed.   "call context be readonly" must be inside a staticcall?

Reply:
describe it here to make it more clear.